app/plugins/modules/editor/usage.js

```diff
@@ -18,6 +18,13 @@
 const all = ['compose', 'new', 'edit']
 const allExcept = cmd => all.filter(_ => _ !== cmd)
 
+/** optional arguments for new and compose commands */
+const optional = [{ name: '--kind', alias: '-k', docs: 'The OpenWhisk kind of the new action',
+                    allowed: ['nodejs', 'python', 'php', 'swift'],
```

Comment:
can the kinds also be configured per build (or derived from the openwhisk deployment which would be a more accurate reflection of the supported kinds that you point the shell to... of course that makes the configuration specific to an api host).